### PR TITLE
Fix building on MinGW-w64

### PR DESCRIPTION
**Pull request by MehdiChinoune**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -199,12 +199,13 @@ endif()
...
 if(WIN32)
-  if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  if(MSVC AND NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     message(FATAL_ERROR
       "Your compiler is not supported; install Clang from Visual Studio Installer and re-run cmake with '-T clangcl'")
   endif()
 
-  target_compile_definitions(mold PRIVATE NOGDI NOMINMAX)
+  target_compile_definitions(mold PRIVATE NOGDI NOMINMAX _WIN32_WINNT=0xA00)
+  target_link_libraries(mold PRIVATE bcrypt)
```
Comment (rui314) [inline CMakeLists.txt:208]: Can this guard with `if mingw`?

Comment (MehdiChinoune) [reply CMakeLists.txt]: OK

**Comment by rui314** [comment]
Please sign your commit as explained in https://github.com/rui314/mold/blob/main/CONTRIBUTING.md

**Comment by MehdiChinoune** [comment]
> Please sign your commit as explained in https://github.com/rui314/mold/blob/main/CONTRIBUTING.md

OK!

**Pull request merged**